atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RowColumnRangeExtractor.java

```diff
@@ -104,17 +111,24 @@ static RowColumnRangeResult extract(
             rowsToRawColumnCount.put(row, columns.size());
             for (ColumnOrSuperColumn c : columns) {
                 Pair<byte[], Long> pair = CassandraKeyValueServices.decomposeName(c.getColumn());
+                // Column name
+                TraceStatistics.incBytesRead(pair.lhSide.length);
+                // Column value
+                TraceStatistics.incBytesRead(c.getColumn().getValue().length);
+
                 long ts = pair.rhSide;
                 if (ts < startTs) {
                     Cell cell = Cell.create(row, pair.lhSide);
                     LinkedHashMap<Cell, Value> cellToValue =
                             collector.computeIfAbsent(row, _b -> new LinkedHashMap<>(1));
                     if (cellToValue.containsKey(cell)) {
+                        TraceStatistics.incEmptyValues(1);
                         notLatestVisibleValueCellFilterCounter.get().inc();
```

Comment:
I think the empty values trace statistic is badly named? @jeremyk-91 which should we be tracking (empty values vs not-latest ts?).

Reply:
Ah. So they're probably worth tracking in their own right. You're right that in this case the value is not-latest as opposed to empty.

- Empty values are Atlas tombstones. This can be because Sweep isn't fully up to date on a thoroughly swept table where the user has deleted something, or because Atlas is trying to range scan a conservatively swept table which has had a large number of deletions. (We've seen this latter case on internal navigation product.)
- Not latest TS values are old versions of a cell that haven't been swept away yet (possibly because sweep is slow, or because there is a long running transaction that could read said version).

A large number of either is generally not a good thing, though they do catch different things, so agree that keeping both should be useful.